cmd/iop/shared.go

```diff
@@ -77,12 +77,13 @@ func genManifests(args *rootArgs) (name.ManifestMap, error) {
 	// Start with unmarshaling and validating the user CR (which is an overlay on the base profile).
 	overlayICPS := &v1alpha2.IstioControlPlaneSpec{}
 	if err := util.UnmarshalWithJSONPB(overlayYAML, overlayICPS); err != nil {
-		log.Errorf("Could not unmarshal the overlay YAML from file: %s", overlayFilenameLog)
-		return nil, err
+		e := fmt.Errorf("could not unmarshal the overlay YAML from file: %s, caused by: %v",
+			overlayFilenameLog, err)
+		return nil, e
```

Comment:
It's more idiomatic to just write
return nil, fmt.Errorf(...), here and elsewhere.

Reply:
Sure.